### PR TITLE
Migrate html2pdf to composer

### PR DESCRIPTION
**Pull request by Wakie87**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,6 +10,8 @@
         "source": "http://github.com/openemr/openemr"
     },
 	"require": {
+        "spipu/html2pdf": "4.5.0",
+        "setasign/fpdi-tcpdf": "1.6.*"
```
Comment (bradymiller) [inline composer.json]: Another stupid question. How does this know to get specific versions of fpdi and tcpdf from tecnickcom github repo?

**Comment by bradymiller** [comment]
This is a bit complicated.

Would need to deal with a couple issues that required customization of this code described here:
http://www.open-emr.org/wiki/index.php/Html2pdf#OpenEMR_4.3.1_and_above

Step 6. Note sure how this can be done without modifying the code.

Step 7. I think we are ok there since it's getting autoloaded already. correct?

Step 9. What are you thoughts on this?

Step 11. Note that this bug has been fixed in upstream html2pdf and is hopefully in version 4.6.0 (note this was released on the same day the bug was fixed: https://github.com/spipu/html2pdf/pull/96 ). If the bug is fixed in html2pdf, will need to change the call in in the reports script since RTL is not a parameter of the function initiation and is instead set after creating the class (which I would be happy to do).

**Comment by bradymiller** [comment]
Hopefully we can pass above hurdles without needing to modify code. If we need to modify it in the end, though, this sounds like it would merit our own repos for these. Would that be correct?

**Comment by Wakie87** [comment]
Step 6. Why are we doing this? Does everything work without doing this? 
Step 7. We are good because of the autoloading
Step 9. By including `setasign/fpdi-tcpdf": "1.6.1"` in composer.json it evaluate the dependencies automatically.
Step 11. By requiring a newer version in the composer.json file we can overcome this.

**Comment by bradymiller** [comment]
Regarding step 6, could try without it, but I am pretty sure this was done for a reason (guessing that it will not bundle up the pdf files correctly into the pdf output if not set). Again, though, can leave it out and see what happens on testing.

**Comment by bradymiller** [comment]
Hi,

Here's the codebase fix to support RTL in the html2pdf version 4.6.0:
https://github.com/bradymiller/openemr/commit/c75c00a2e2d4dcbaa5a74458aed74a302cb95f12

Recommend bringing this into your branch and then we can begin to test it (on initial testing I am seeing some weirdness in pdf handling, but plan to do more testing(ie. there is a concern step 6 may be needed)).

-brady

**Comment by bradymiller** [comment]
Ongoing work on this is here(this is on top of most recent codebase):
https://github.com/bradymiller/openemr/commits/testing-migrate-html2pdf_4

Note that need to edit the package in 1 place for it to work:
https://github.com/bradymiller/openemr/commit/2fbf72ef49cab51c81f6ca980fbfa8e795115014

Need to figure out best strategy to do this??

Also, when this is complete, should run the composer install again and use build.xml to remove the examples directory from the html2pdf package.

**Comment by Wakie87** [comment]
To make things easier, Have removed the HTML2PDF library and used the already included TCPDF and FPDI. Will need some more testing.

**Comment by sunsetsystems** [comment]
You can't just zap the customizations.  Did you read library/html2pdf/README_OpenEMR.txt?

**Comment by bradymiller** [comment]
Hi Rod,
In my comment above, the only customization needed to get it to work is there( take from http://www.open-emr.org/wiki/index.php/Html2pdf#OpenEMR_4.3.1_and_above which is basically library/html2pdf/README_OpenEMR.txt ). Looks like Scott is working on a way to avoid needing that customization and this is likely experimental.
-brady

**Comment by Wakie87** [comment]
See https://github.com/openemr/openemr/pull/190

**Pull request closed without merging**